packages/fast-usdc-contract/src/fast-usdc.contract.ts

```diff
@@ -59,26 +60,38 @@ export const NOBLE_ICA_BAGGAGE_KEY = 'nobleICA';
 
 export const meta = {
   customTermsShape: FastUSDCTermsShape,
-  privateArgsShape: {
-    // @ts-expect-error TypedPattern not recognized as record
-    ...OrchestrationPowersShape,
-    assetInfo: M.arrayOf([DenomShape, DenomDetailShape]),
-    chainInfo: M.recordOf(M.string(), CosmosChainInfoShapeV1),
-    feeConfig: FeeConfigShape,
-    marshaller: M.remotable(),
-    poolMetricsNode: M.remotable(),
-  },
-} as ContractMeta<typeof start>;
+  privateArgsShape: M.splitRecord(
+    {
+      // @ts-expect-error TypedPattern not recognized as record
+      ...OrchestrationPowersShape,
+      assetInfo: M.arrayOf([DenomShape, DenomDetailShape]),
+      chainInfo: M.recordOf(M.string(), CosmosChainInfoShapeV1),
+      /** default fee config for all destinations */
+      feeConfig: FeeConfigShape,
+      marshaller: M.remotable(),
+      poolMetricsNode: M.remotable(),
+    },
+    {
+      /**
+       * if specified for a {@link CaipChainId}, supersedes the default `FeeConfig` when
+       * the specified chain is the destination
+       */
+      customFeeConfigs: M.recordOf(M.string(), FeeConfigShape),
+    },
+  ), // XXX ContractMeta doesn't understand splitRecord?
```

Comment:
all the changes in this file go away with a unified `feeConfig`

Reply:
Good suggestion, done